lib/client.js

```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+const nodeVault = require('node-vault')
+
+class HashiCorpClient {
+  /**
+   * @param options {{mountPoint?: string, vaultOptions: import("node-vault").VaultOptions}}
+   */
+  constructor({ mountPoint = 'secret', vaultOptions } = {}) {
```

Comment:
this does not do what you think it's doing, or anyway it's really hard to read. I would recommend to move the default down.

Reply:
My understanding is this

- allows the constructor argument to be undefined and still initialise mountPoint to 'secret'
- allows the constructor argument to be an object without a mountPoint key and still initialise mountPoint to 'secret'

If this is doing something else, or has any unwanted effects, please let me know.

Personally I like this syntax but that's probably just because I'm used to it, I have moved the deconstruction to the line below.